packages/debugger/lib/data/sagas/index.js

```diff
@@ -1286,6 +1295,64 @@ function* requestCode(address) {
   }
 }
 
+//NOTE: just like requestCode, this can also add to the codex!
+//yes, this is also weird.
+function* requestStorage(slot) {
+  //since the debugger only requests storage when it doesn't know it,
+  //we won't bother here with a check regarding whether we already
+  //know this storage or not; we can assume that we don't
+  //NOTE: this is also assuming that no individual variable ever requires
+  //reading the same storage value twice in the course of decoding!
+  //this assumption is presently true, but if it ever becomes false, then it
+  //will become necessary to add that check here (not that the result would
+  //be *wrong* without such a check, but it would cause unnecessary network
+  //requests)
+  const address = yield select(data.current.address);
+  const blockHash = yield select(data.views.blockHash); //cannot use number here!
+  const txIndex = yield select(data.views.txIndex);
+  const word = yield* web3.obtainStorage(address, slot, blockHash, txIndex);
```

Comment:
what happens if this returns an error?

Reply:
You mean if this throws?  Well, then we throw.  I don't know how we'd recover from an error here!

If your question is, what happens if this throws for the specific reason that `debug_storageRangeAt` isn't supported, rather than for any other reason, then the answer is, that won't happen because if `debug_storageRangeAt` isn't supported we won't end up on this code path!

Reply:
I think I was getting myself confused. `web3.obtainStorage` apparently throws... for some reason I thought I saw that it returned its errors.